spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+  RSpec.describe "Recipes Show" do
+  before(:each) do
+    @recipe_1 = Recipe.create(name: "pie", complexity: 5, genre: "baked")
+
+    @ingredient_1 = Ingredient.create(name: "pepper", cost: 3)
+    @ingredient_2 = Ingredient.create(name: "salt", cost: 2)
+
+    RecipeIngredient.create(recipe: @recipe_1, ingredient: @ingredient_1)
+    RecipeIngredient.create(recipe: @recipe_1, ingredient: @ingredient_2)
+  end
+
+  describe "When I visit '/recipes/:id'" do
+    it "I see a list of all the ingredients including their name and cost" do
+      visit "/recipes/#{@recipe_1.id}"
```

Comment:
I would probably break this test up. US 2 asks for
```
As a visitor,
When I visit '/recipes/:id',
Then I see the recipe's name, complexity and genre,
and I see a list of the names of the ingredients for the recipe
```
It looks like you have the name, complexity, and genre but don't have the list of ingredient names. Then you have `total cost`. So I would break it up into 3 tests:
1. name, complexity, genre
2. list of names of all the ingredients for the recipe
3. total cost of all the ingredients in the recipe